src/main.rs

```diff
@@ -42,8 +42,9 @@ global_asm!(include_str!("video_mode/vga_320x200.s"));
 global_asm!(include_str!("video_mode/vga_text_80x25.s"));
 
 unsafe fn context_switch(boot_info: VirtAddr, entry_point: VirtAddr, stack_pointer: VirtAddr) -> ! {
-    llvm_asm!("call $1; ${:private}.spin.${:uid}: jmp ${:private}.spin.${:uid}" ::
-         "{rsp}"(stack_pointer), "r"(entry_point), "{rdi}"(boot_info) :: "intel");
+    asm!("movq {1}, %rsp; callq *{0}; 0: jmp 0b",
```

Comment:
Unfortunately LLVM parses the `0b` as binary literal when using intel syntax. Using a named label is not allowed either as they can cause conflicts when rustc or LLVM decides to duplicate an inline asm block because it for example inlines it. For this reason I had to switch to AT&T syntax.

Reply:
Have you tried using 2 or larger as label name? https://github.com/rust-lang/rust/issues/76704#issuecomment-815351643 seems to indicate that only 0 and 1 are problematic.